staging/src/k8s.io/code-generator/cmd/register-gen/generators/register_external.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generators
+
+import (
+	"io"
+	"text/template"
+
+	clientgentypes "k8s.io/code-generator/cmd/client-gen/types"
+	"k8s.io/gengo/generator"
+	"k8s.io/gengo/types"
+)
+
+type registerExternalGenerator struct {
+	generator.DefaultGen
+	gv              clientgentypes.GroupVersion
+	typesToGenerate []*types.Type
+}
+
+var _ generator.Generator = &registerExternalGenerator{}
+
+func (g *registerExternalGenerator) Filter(_ *generator.Context, _ *types.Type) bool {
+	return false
+}
+
+func (g *registerExternalGenerator) Imports(c *generator.Context) []string {
+	imports := []string{
+		"metav1 \"k8s.io/apimachinery/pkg/apis/meta/v1\"",
```

Comment:
@sttts this strikes me as odd.  Do you have a better example?

Reply:
This should be automatic if the import tracker is used throughout the generated code. It should.